blobs.go

```diff
@@ -0,0 +1,71 @@
+package distribution
+
+import (
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/docker/distribution/digest"
+	"golang.org/x/net/context"
+)
+
+// Blob describes a readable and seekable binary object. Most registry
+// objects, including layers and manifests can be accessible by digest in a
+// BlobService.
+type Blob interface {
+	Describable
+	io.ReaderAt
+	io.Closer
+
+	Reader() (ReadSeekCloser, error)
```

Comment:
@dmcgowan @tibor What do you think about removing this method completely and just supporting io.ReaderAt on blob. It allows us to abstract blobs as a series of distributed chunks and defers the `io.Reader` implementation to the caller (ie `NewReadSeeker(length int64, blob io.ReaderAt) (io.ReadCloser, error)`).